app/queues/dns/dns-queue.server.ts

```diff
@@ -30,9 +63,36 @@ function init() {
           status: DnsRecordStatus.error,
         });
         logger.debug(`DNS job failed, synced record id=${dnsRecordId} to error status`, job.data);
+        await sendDnsRecordStatusUpdateNotification(dnsRecordId, DnsRecordStatus.error);
+      } catch (err) {
+        logger.error(
+          `DNS job failed, unable to sync record id=${dnsRecordId} to error status and/or add notification`,
+          job.data,
+          err
+        );
+      }
+    }
+  });
+
+  /**
+   * When a DNS job completes, update the status in the db, and add email notification job
+   */
+  dnsQueueEvents.on('completed', async ({ jobId }) => {
+    const job = await dnsQueue.getJob(jobId);
+    if (job) {
+      const { dnsRecordId } = job.data;
+      try {
+        await updateDnsRecordById(dnsRecordId, {
```

Comment:
I already do this in the worker, see https://github.com/DevelopingSpace/starchart/blob/main/app/queues/dns/dns-worker.server.ts#L128-L147

Let's only send the notification here.

I the `fail` case I have to update the db, because I can't deal with failure in the worker itself due to retries.

Reply:
My bad; let me fix this.